meeteval/wer/__main__.py

```diff
@@ -121,56 +112,18 @@ def _glob(pathname):
     # which one we have and load it
     hypothesis = _load_hypothesis(hypothesis_paths)
 
-    # Group by example IDs
-    reference = reference.grouped_by_filename()
-    hypothesis = hypothesis.grouped_by_filename()
-
+    # Filter lines with regex based on filename
     if regex:
         r = re.compile(regex)
-        def filter(grouped):
-            keys = grouped.keys()
-            grouped = {k: v for k, v in grouped.items() if r.fullmatch(k)}
-            assert grouped, (regex, keys, 'Found nothing')
-            return grouped
+
+        def filter(stm):
+            filenames = stm.filenames()
+            filtered_filenames = [f for f in filenames if r.fullmatch(f)]
+            assert filtered_filenames, (regex, filenames, 'Found nothing')
+            return stm.filter(lambda l: l.filename in filtered_filenames)
```

Comment:
Does the following work?
```python
        def filter(stm):
            new = stm.filter(lambda line: r.fullmatch(line.filename))
            assert len(new), (regex, stm.filenames(), 'Found nothing')
```
It's cleaner, but maybe a bit more expensive.

Reply:
I intentionally didn't run the match on all lines since there can be many lines with the same filename. But I don't know how much of a difference this makes

Reply:
Ah, ok. I thought the code only translated from dict to stm.